### PR TITLE
Add commit, plan, and execute timing information to `QueryResult` and `Result`

### PR DESCRIPTION
**Pull request by iheanyi**
## Description

This pull request adds various timing information for plan, execute, and commit to `QueryResult`. This is valuable information that callers communicating with `vtgate` can use for observability information and also having an accurate idea of how long various statements took, without the network overhead.

## Related Issue(s)
N/A

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: proto/query.proto
Comment (mattrobenolt) [inline proto/query.proto:404]: Would it make sense to wrap this in a `QueryStats` message instead?

So something like: 

```proto
message QueryStats {
  vttime.Duration execute = 1;
  vttime.Duration plan = 2;
  vttime.Duration commit = 3;
```

Then just add:

```
message QueryResult {
 ...
  QueryStats stats = x;
}```

Seems a bit better to wrap this in it's own Message imo, but I don't have strong opinions about this.

**Review by frouioui — APPROVED** [review]
This change looks good to me. I will wait for a second approval to merge this.

File: proto/query.proto
Comment (iheanyi) [reply proto/query.proto]: Okay, this has been addressed.

**Review by vmg — CHANGES_REQUESTED** [review]
I don't think this is the right design for what we're trying to do here. The `QueryResult` is a core proto message for all query serving, i.e. it's using as an intermediate result for every stream event during stream executes, in the vstream, etc, and in many places where having `QueryStats` either doesn't make sense or is redundant.

I would suggest rethinking this to accomplish the goal we have without cluttering this core message. If we're looking for the runtime stats of queries when they're `Executed`, why not return a `QueryStats` result via GRPC metadata instead of embedding a new field?

**Pull request closed without merging**
